internal/device/darwin.go

```diff
@@ -8,9 +8,11 @@ import (
 	"log"
 	"os/exec"
 	"regexp"
+	"strconv"
+	"strings"
 )
 
-const POWER_REGEX = `[\d]{1,2}%`
+const POWER_REGEX = `[\d]{1,3}%;\s[\S]+[\s(\S)+]*;`
 
 func GetPower() string {
 	cmd := exec.Command("pmset", "-g", "ps")
```

Comment:
In general I don't like it when I'm forced to shell out to some command to get the information I want, and then parse its output; that's quite fragile and vulnerable, and also relies on having this command available (which it wouldn't be in a container, for example).

I'm sure you feel the same, so have you looked into any other ways to obtain this data, such as a system API, for example? What does the `pmset` program itself do? Presumably just make system calls; we could make the same calls, couldn't we?

Reply:
I had found this on  [stackoverflow ](https://stackoverflow.com/questions/272552/battery-status-api-in-macos) which led me to this doc in [Apple](https://developer.apple.com/documentation/iokit/iopowersources_h). 

I don't know how to use it.  Is there any way using which we can use it?

Reply:
Yes, it looks like Apple doesn't provide a Go SDK for this stuff, which is annoying. The closest things I can find to anyone talking directly to MacOS in Go are:

* https://github.com/go-darwin
* https://gist.github.com/nicolai86/fd720c9a660188fe29d7ac7a810f96f3

neither of which is particularly helpful to you. Writing cgo wrappers for what you want to do is a whole lot of effort and not really worth it.

So if we have to drive `pmset`, then so be it! But let's try to wrap `pmset`'s functionality with a general-purpose Go library. It should be able to do both what _you_ need, for this specific tool, and ideally also be helpful to other people writing Go tools around Mac power management.

Reply:
I have created a wrapper over pmset (Sent you invite for the same) and its  corresponding [PR](https://github.com/A9u/urja/pull/1/)